specification/sbi_atee.adoc

```diff
@@ -10,8 +39,1296 @@ shown in the figure above, the TEE Runtime (operating in VS-mode) supports
+.TEE Host Get TSM Info
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code              | Description
+| SBI_SUCCESS             | The operation completed successfully.
+| SBI_ERR_INVALID_ADDRESS | `tsm_info_address` was invalid.
+| SBI_ERR_INVALID_PARAM   | tsm_info_len was insufficient.
+| SBI_ERR_FAILED          | The operation failed for unknown reasons.
+|===
+
+A list of possible TSM states and the associated semantics appears below (TBD: States for TSM update).
+
+[#table_tsm_states]
+.TSM States
+[%header,%autowidth]
+|===
+| TSM State          | Meaning
+
+| TSM_NOT_LOADED     | TSM has not been loaded on this platform.
+| TSM_LOADED         | TSM has been loaded, but has not yet been initialized.
+| TSM_READY          | TSM has been loaded & initialized, and is ready to accept ECALLs.
+|===
+
+=== Function: TEE Host Convert Pages (FID #1)
+[source, C]
+-----
+struct sbiret sbi_tee_host_convert_pages(unsigned long base_page_address,
+                                         unsigned long num_pages);
+
+-----
+
+Begins the process of converting `num_pages` of non-confidential memory starting
+at `base_page_address` to confidential-memory. On success, pages can be assigned
+to TVMs only following subsequent calls to `sbi_tee_host_global_fence()` and
+`sbi_tee_host_local_fence()` that complete the conversion process. The implied
+page size is 4KiB.
+
+The `base_page_address` must be page-aligned.
+
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_host_convert_pages_errors]
+.TEE Host Convert Pages
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code              | Description
+| SBI_SUCCESS             | The operation completed successfully.
+| SBI_ERR_INVALID_ADDRESS | `base_page_address` was invalid.
+| SBI_ERR_INVALID_PARAM   | `num_pages` was invalid.
+| SBI_ERR_FAILED          | The operation failed for unknown reasons.
+|===
+
+
+=== Function: TEE Host Reclaim Pages (FID #2)
+[source, C]
+-------
+struct sbiret sbi_tee_host_reclaim_pages(unsigned long base_page_address,
+                                         unsigned long num_pages);
+
+-------
+Reclaims `num_pages` of confidential memory starting at `base_page_address`.
+The pages must not be currently assigned to an active TVM. The implied page
+size is 4KiB.
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_tee_tsm_reclaim_pages_errors]
+.TEE Host Reclaim Pages
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code              | Description
+| SBI_SUCCESS             | The operation completed successfully.
+| SBI_ERR_INVALID_ADDRESS | `base_page_address` was invalid.
+| SBI_ERR_INVALID_PARAM   | `num_pages` was invalid.
+| SBI_ERR_FAILED          | The operation failed for unknown reasons.
+|===
+
+=== Function: TEE Host Initiate Global Fence (FID #3)
+[source, C]
+-----
+struct sbiret sbi_tee_host_global_fence(void);
+-----
```

Comment:
nit: I would drop the "global" here. It's confusing in the presence of local_fence below.